internal/framework/services/s3/resource_bucket_cors_configuration_test.go

```diff
@@ -0,0 +1,188 @@
+					resource.TestCheckResourceAttr(name, "cors_rule.0.allowed_methods.0", "POST"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.allowed_methods.1", "PUT"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.allowed_origins.#", "1"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.allowed_origins.0", "https://s3-website-test.hashicorp.com"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.expose_headers.#", "1"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.expose_headers.0", "ETag"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.max_age_seconds", "3000"),
+					resource.TestCheckResourceAttr(name, "cors_rule.0.id", "1234"),
+					resource.TestCheckResourceAttr(name, "cors_rule.1.allowed_headers.#", "0"),
+					resource.TestCheckResourceAttr(name, "cors_rule.1.allowed_methods.#", "2"),
+					resource.TestCheckResourceAttr(name, "cors_rule.1.allowed_methods.0", "DELETE"),
+					resource.TestCheckResourceAttr(name, "cors_rule.1.allowed_methods.1", "GET"),
+					resource.TestCheckResourceAttr(name, "cors_rule.1.allowed_origins.#", "1"),
+					resource.TestCheckResourceAttr(name, "cors_rule.1.allowed_origins.0", "https://s3.ionoscloud.com"),
+				),
+			},
+		},
+	})
+}
+
+func testAccBucketCORSConfigurationConfig_base(bucketName string) string {
+	return fmt.Sprintf(`
+resource "ionoscloud_s3_bucket" "test" {
+  name = %[1]q
+  region = "eu-central-3"
+}
+`, bucketName)
+}
+
+func testAccBucketCORSConfigurationConfig_basic(bucketName string) string {
+	return utils.ConfigCompose(testAccBucketCORSConfigurationConfig_base(bucketName), fmt.Sprintf(`
+resource "ionoscloud_s3_bucket_cors_configuration" "test" {
+  bucket = ionoscloud_s3_bucket.test.name
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["PUT", "POST"]
+    allowed_origins = ["https://s3-website-test.hashicorp.com"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+    id = 1234
+  }
+}
+`))
+}
+
+func testAccBucketCORSConfigurationConfig_multiple(bucketName string) string {
+	return utils.ConfigCompose(testAccBucketCORSConfigurationConfig_base(bucketName), fmt.Sprintf(`
+resource "ionoscloud_s3_bucket_cors_configuration" "test" {
+  bucket = ionoscloud_s3_bucket.test.name
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["PUT", "POST"]
+    allowed_origins = ["https://s3-website-test.hashicorp.com"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+    id = 1234
+  }
+
+ cors_rule {
+    allowed_methods = ["DELETE", "GET"]
+    allowed_origins = ["https://s3.ionoscloud.com"]
+ }
+}
+`))
+}
+
+func testAccCheckBucketCORSConfigurationDestroy(s *terraform.State) error {
+	client, err := acctest.S3Client()
+	if err != nil {
+		return err
+	}
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ionoscloud_s3_bucket_cors_configuration" {
+			continue
+		}
+
+		if rs.Primary.Attributes["bucket"] != "" {
```

Comment:
you can just return nil if rs.Primary.Attributes["bucket"] == ""

Reply:
why?

Reply:
less levels of nesting. easier to read. doesn't matter that much.